pkg/manager/member/pump_member_manager.go

```diff
@@ -0,0 +1,475 @@
+// Copyright 2019. PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package member
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"text/template"
+
+	"github.com/BurntSushi/toml"
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/controller"
+	"github.com/pingcap/tidb-operator/pkg/label"
+	"github.com/pingcap/tidb-operator/pkg/manager"
+	"github.com/pingcap/tidb-operator/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	v1 "k8s.io/client-go/listers/apps/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog"
+)
+
+// pumpStartScriptTpl is the template string of pump start script
+// Note: changing this will cause a rolling-update of pump cluster
+var pumpStartScriptTpl = template.Must(template.New("pump-start-script").Parse(`set -euo pipefail
+
+/pump \
+-pd-urls={{ .Scheme }}://{{ .ClusterName }}-pd:2379 \
+-L={{ .LogLevel }} \
+-advertise-addr=` + "`" + `echo ${HOSTNAME}` + "`" + `.{{ .ClusterName }}-pump:8250 \
```

Comment:
this is strange, does this work? `-advertise-addr={{.PodName}}.{{ .ClusterName }}-pump:8250`
we don't need to pass HOSTNAME environment for sts pod, because the pod address is predictable

Reply:
Use the `HOSTNAME` environment deliberately to keep backward compatibility with the helm chart, otherwise the pump statefulset will get rolling-update if we set the new env var `POD_NAME`.